syntax = "proto3";


import "vo.proto";


message ProjectCreated{
    string project_id = 1;
}





// DATASET VERSION --------------------------------------------


message DatasetCreated{
    string project_id = 1;
    string dataset_id = 2;


    DatasetMetadataDelta meta = 4;
}

message DatasetUpdated {
    string project_id = 1;
    string dataset_id = 2;

    DatasetMetadataDelta meta = 3;
}



// META ---------------------------------------

enum Type {
    None = 0;
    Event_ProjectCreated = 1;
    Event_DatasetCreated = 2;
    Event_DatasetUpdated = 3;
}


message Event {
    bytes Body = 1;
    Type Type = 2;
}


